File: 0668-kth-smallest-number-in-multiplication-table/0668-kth-smallest-number-in-multiplication-table.py
class Solution:
    def funToCount(self, mid, m, n):
        count = 0
        for i in range(1, m+1):
            count += min(mid//i, n)
        return count
        
    def findKthNumber(self, m: int, n: int, k: int) -> int:
        lo, hi = 1, m*n
        
        while lo < hi:
            mid = (lo + hi)//2
            count = self.funToCount(mid, m, n)
            
            if count >= k: hi = mid
            else: lo = mid+1
        return lo

File: 0668-kth-smallest-number-in-multiplication-table/0668-kth-smallest-number-in-multiplication-table.java
class Solution {
    int countNoLessThanMid(int mid, int m, int n)
    {
        int count = 0;
        for(int i=1; i<=m; i++)
        {
            count += Math.min(mid/i, n); 
        }
        return count;
    }
    public int findKthNumber(int m, int n, int k) {
        int lo = 0;
        int hi = m*n;
        while (lo < hi) 
        {
            int mid = (lo+hi)/2;
            int count = countNoLessThanMid(mid, m, n);
            
            if (count >= k) 
                hi = mid;
            else  
                lo = mid+1;
        }
        return lo;
    }
}